app/src/androidTest/java/com/lahtinen/apps/flextime/MainActivityTest.kt

```diff
@@ -0,0 +1,117 @@
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.rule.ActivityTestRule
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class MainActivityTest {
+
+    /**
+     * NOTE
+     * You need to disable Animations and transitions on the target device.
+     * https://stackoverflow.com/questions/44005338/android-espresso-performexception
+     */
+
+    @get:Rule
+    val activityRule = ActivityTestRule(MainActivity::class.java)
+
+    @Test
+    fun addTime() {
+        reset()
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("04:30"))
+
+        onView(withId(R.id.btAdd)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("4h30m")))
+
+        //picker is reset so nothing should happen
+        onView(withId(R.id.btAdd)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("4h30m")))
+
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("02:15"))
+        onView(withId(R.id.btAdd)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("6h45m")))
+    }
+
+    @Test
+    fun subtractTime() {
+        reset()
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("00:30"))
+        onView(withId(R.id.btSubtract)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("-0h30m")))
+
+        //picker is reset so nothing should happen
+        onView(withId(R.id.btSubtract)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("-0h30m")))
+
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("00:45"))
+        onView(withId(R.id.btSubtract)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("-1h15m")))
+    }
+
+    @Test
+    fun combineAddAndSubtractTime() {
+        reset()
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("00:30"))
+        onView(withId(R.id.btSubtract)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("-0h30m")))
+
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("00:45"))
+        onView(withId(R.id.btAdd)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("0h15m")))
+    }
+
+    @Test
+    fun onResetClick_resetFlexTime() {
+        reset()
+
+        onView(withId(R.id.timePicker)).perform(NumberPickerSetter("04:15"))
+        onView(withId(R.id.btSubtract)).perform(click())
+        onView(withId(R.id.tvTime)).check(matches(withText("-4h15m")))
```

Comment:
I don't think this assertion is necessary since it's covered by other tests. Generally, only assert what the test is supposed to verify.